snpe/notebooks/non-marketplace_simulations.ipynb

```diff
@@ -0,0 +1,446 @@
+    "    previous_rating_measure: str,\n",
+    "    min_reviews_for_herding: int,\n",
+    "    herding_differentiating_measure: str,\n",
+    "    simulation_class: int,\n",
+    ") -> None:\n",
+    "    assert simulation_class in [\n",
+    "        \"singlerho\",\n",
+    "        \"doublerho\",\n",
+    "        \"herding\",\n",
+    "        \"doubleherding\",\n",
+    "    ], \" Can only use singlerho/doublerho/herding/doubleherding as simulation classes, please enter a valid class.\"\n",
+    "    params = {\n",
+    "        \"review_prior\": review_prior,\n",
+    "        \"tendency_to_rate\": tendency_to_rate,\n",
+    "        \"simulation_type\": simulation_type,\n",
+    "        \"previous_rating_measure\": previous_rating_measure,\n",
+    "    }\n",
+    "\n",
+    "    params_double = params.copy()\n",
+    "    params_herding = params.copy()\n",
+    "    params_herding.update(\n",
+    "        {\n",
+    "            \"previous_rating_measure\": previous_rating_measure,\n",
+    "            \"min_reviews_for_herding\": min_reviews_for_herding,\n",
+    "        }\n",
+    "    )\n",
+    "\n",
+    "    params_double_herding = params_herding.copy()\n",
+    "    params_double_herding.update(\n",
+    "        {\"herding_differentiating_measure\": herding_differentiating_measure}\n",
+    "    )\n",
+    "\n",
+    "    simulation_classes = {\n",
+    "        \"singlerho\": simulator_class.SingleRhoSimulator(params),\n",
+    "        \"doublerho\": simulator_class.DoubleRhoSimulator(params_double),\n",
+    "        \"herding\": simulator_class.HerdingSimulator(params_herding),\n",
+    "        \"doubleherding\": simulator_class.DoubleHerdingSimulator(params_double_herding),\n",
+    "    }\n",
+    "\n",
+    "    simulator = simulation_classes.get(simulation_class)\n",
+    "    print(\"Simulation type\" + str(simulation_classes.get(simulation_class)))\n",
+    "    simulator.simulate(num_simulations=num_simulations)\n",
+    "    simulator.save_simulations(ARTIFACT_PATH)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e680cde0",
+   "metadata": {},
+   "source": [
+    "### D. Running the simulation"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "6df62b31",
+   "metadata": {},
+   "source": [
+    "Finally we can run the simulation by calling the \"generate_and_save\" function providing the desired parameters. In the example below a Double herding simulation of the timeseries type is carried out for ten consumers. Five ratings (one for each rating value) are pre-loaded, the tendency to rate is set at 5%, the mode (of all previous ratings) is taken as the reference metric for herding while the threshold for the doubleherding is determined by the mean (of al previous ratings). Lastly, 5 is the minimum number of previous reviews required for herding. (Given that we are already providing 5 ratings to begin with, herding will be possible from the start of the simulation). "
```

Comment:
Awesome, this is really good work! Just 1 final suggestion: can you put in an example run of each of the 4 different simulators (just like you did for the DoubleHerdingSimulator) so that any future user knows that the function you wrote works for all 4?

Reply:
Thank you for the comments! I've just included the examples, moved the notebook to the newly created snpe/tutorials directory and commited all of these modifications

Reply:
Thanks for explaining, no I think this looks good enough for the purposes of a tutorial since that function only uses the underlying classes and doesn't change anything about the classes. I think it looks good for merging, will just tag @shrabasteebanerjee to keep her informed that we are merging.